venues/OpenReview.net/Anonymous_Preprint/python/deanonymize.py

```diff
@@ -0,0 +1,50 @@
+import openreview
+import argparse
+
+def deanonymize(mask, original):
+    '''
+    Deanonymizes an OpenReview Anonymous Archive submission by:
+    1) Removing `authors` and `authorids` from the Blind_Submission reference
+    2) Replacing the Authors field of the bibtex with the authors from the Original
+    3) Placing the lowercased last name of the first author in the bibtex ID
+
+    Returns the deanonymized reference to be posted.
+    '''
+
+    mask.content.pop('authors')
+    mask.content.pop('authorids')
+    mask.content['_bibtex'] = mask.content['_bibtex'].replace(
+        '{Anonymous}',
+        ', '.join(original.content['authors']))
+
+    first_author = original.content['authors'][0]
+    lastname = first_author.split(' ')[-1].lower()
+
+    mask.content['_bibtex'] = mask.content['_bibtex'].replace(
```

Comment:
not sure if this is really a good way of doing this replacement... let me know what you think

Reply:
We need to move the bibtex generation to our backend soon.